src/App.tsx

```diff
@@ -15,48 +16,106 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Default = 'default',
+  Alphabet = 'alphabet',
+  Length = 'length',
+}
+
+interface SortFields {
+  sortField: SortType,
+  isReverse: boolean,
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, isReverse }: SortFields,
+): string[] {
+  const preparedGoods = [...goods];
+
+  if (sortField !== SortType.Default) {
+    switch (sortField) {
+      case SortType.Alphabet:
+        return preparedGoods.sort((goodA, goodB) => goodA.localeCompare(goodB));
+      case SortType.Length:
+        return preparedGoods
+          .sort((goodA, goodB) => goodA.length - goodB.length);
+      default:
+        return preparedGoods;
+    }
+  }
+
+  if (isReverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.Default);
+  const [isReverse, setIsReverse] = useState<boolean>(false);
+  const visibleGoods: string[] = getPreparedGoods(goodsFromServer,
+    { sortField, isReverse });
+
+  const handlerOnReset = () => {
```

Comment:
```suggestion
  const handleReset = () => {
```
Prefix `handle` for a handler and the prefix `on` if it's a prop. You don't need to combine them.
You can read about convention [here](https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364)